crux-test/test/crux/fixtures/doc_store.clj

```diff
@@ -0,0 +1,29 @@
+(ns crux.fixtures.doc-store
+  (:require [clojure.tools.logging :as log]
+            [crux.db :as db]
+            [crux.fixtures.api :as apif]
+            [crux.io :as cio])
+  (:import java.io.Closeable))
+
+(defrecord InMemDocumentStore [docs]
+  Closeable
+  (close [_])
+
+  db/DocumentStore
+  (fetch-docs [this ids]
+    (log/debug "Fetching" (cio/pr-edn-str ids))
+    (into {}
+          (for [id ids]
+            [id (get @docs id)])))
+
+  (submit-docs [this id-and-docs]
+    (doseq [[content-hash doc] id-and-docs]
+      (log/debug "Storing" (cio/pr-edn-str content-hash))
+      (swap! docs assoc content-hash doc))))
```

Comment:
nit: `(swap! docs merge id-and-docs)`, I think, will add a commit

Reply:
ah, these aren't maps. happy to merge, but was a bit of a gotcha as `index-docs` takes a map, IIRC?